eZ/Bundle/EzPublishCoreBundle/Command/RemoveVersionsCommand.php

```diff
@@ -0,0 +1,232 @@
+
+use Doctrine\DBAL\Connection;
+use Exception;
+use eZ\Publish\API\Repository\ContentService;
+use eZ\Publish\API\Repository\PermissionResolver;
+use eZ\Publish\API\Repository\UserService;
+use eZ\Publish\API\Repository\Values\Content\VersionInfo;
+use eZ\Publish\Core\Base\Exceptions\InvalidArgumentException;
+use PDO;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\Output;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class RemoveVersionsCommand extends Command
+{
+    const DEFAULT_REPOSITORY_USER = 'admin';
+    const DEFAULT_KEEP = 2;
+
+    const VERSION_DRAFT = 'draft';
+    const VERSION_ARCHIVED = 'archived';
+    const VERSION_ALL = 'all';
+
+    /**
+     * @var \eZ\Publish\API\Repository\UserService
+     */
+    private $userService;
+
+    /**
+     * @var \eZ\Publish\API\Repository\ContentService
+     */
+    private $contentService;
+
+    /**
+     * @var \eZ\Publish\API\Repository\PermissionResolver
+     */
+    private $permissionResolver;
+
+    /**
+     * @var \Doctrine\DBAL\Driver\Connection
+     */
+    private $connection;
+
+    public function __construct(
+        UserService $userService,
+        ContentService $contentService,
+        PermissionResolver $permissionResolver,
+        Connection $connection
+    ) {
+        $this->userService = $userService;
+        $this->contentService = $contentService;
+        $this->permissionResolver = $permissionResolver;
+        $this->connection = $connection;
+
+        parent::__construct();
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        parent::initialize($input, $output);
+
+        $this->permissionResolver->setCurrentUserReference(
+            $this->userService->loadUserByLogin($input->getOption('user'))
+        );
+    }
+
+    protected function configure(): void
+    {
+        $this
+            ->setName('ezplatform:remove-versions')
+            ->setDescription('Remove unwanted content versions. It keeps published version untouched. By default, it keeps also the last archived/draft version.')
+            ->addOption(
+                'status',
+                's',
+                InputOption::VALUE_OPTIONAL,
+                    sprintf(
+                        "Select which version types should be removed: '%s', '%s', '%s'.",
+                        self::VERSION_DRAFT,
+                        self::VERSION_ARCHIVED,
+                        self::VERSION_ALL
+                    ),
+                self::VERSION_ALL
+            )
+            ->addOption(
+                'keep',
+                'k',
+                InputOption::VALUE_OPTIONAL,
+                "Sets number of the most recent versions (both drafts and archived) which won't be removed.",
+                self::DEFAULT_KEEP
```

Comment:
The default should come from Semantic configuration for a repository:
```yml
ezpublish:
    repositories:
        default:
            options:
                default_version_archive_limit: 10
```
See `default_version_archive_limit` [retrieval and usage](https://github.com/ezsystems/ezpublish-kernel/blob/v7.2.3/eZ/Bundle/EzPublishCoreBundle/ApiLoader/RepositoryFactory.php#L113).

_Note: it might not be available in `configure()`. If so, then maybe set it here to null and try to get this value for it either in `execute` or `initialize`._